src/bus/mod.rs

```diff
@@ -46,21 +46,20 @@ enum Listener {
 impl Bus {
     pub async fn for_address(address: &str) -> Result<Self> {
         let mut address = Address::from_str(address)?;
-        let guid: OwnedGuid = match address.guid() {
-            Some(guid) => guid.to_owned().into(),
-            None => {
-                let guid = Guid::generate();
-                address = address.set_guid(guid.clone())?;
 
-                guid.into()
-            }
-        };
         let (listener, auth_mechanism) = match address.transport() {
             #[cfg(unix)]
             Transport::Unix(unix) => {
+                // resolve address specification into address that clients can use
```

Comment:
big nitpick but I prefer the comments to be proper English sentences: `Resolve... can use.` :wink: